### ISSUE TITLE
deployEnvironmentX mutations don't always report errors correctly

### ISSUE THREAD

**Issue by shreddedbacon** [uselagoon/lagoon#2762]
# Describe the bug

When running a `deployEnvironmentX` mutation, if any errors are encountered it doesn't always respond with an actual API error, just a string representation of an error.

Example: 
```
# first update a project to use an invalid build system
mutation updateProject {
  updateProject(input: {
    id: 32, 
    patch: {
      activeSystemsDeploy: "lagoon_kubernetesBuildDeploy"
    }
  }) {
    standbyRoutes
  }
}
# then deploy the latest of an environment within that project
mutation deployLatest {
    deployEnvironmentLatest(input: {
      environment: {
        id:14
      }
    }
  )
}
{
  "data": {
    "deployEnvironmentLatest": "Error: Unknown active system 'lagoon_kubernetesBuildDeploy' for task 'deploy' in for project ci-features-control-k8s"
  }
}
```

Running the same against an environment/project that has no access, we get an actual error from the API.
```
{
  "errors": [
    {
      "message": "Unauthorized",
      "locations": [
        {
          "line": 1412,
          "column": 5
        }
      ],
      "path": [
        "deployEnvironmentLatest"
      ]
    }
  ],
  "data": {
    "deployEnvironmentLatest": null
  }
}
```

## Additional context

This was reported against the CLI, but the actual issue exists in the API (CLI has a possible workaround)
https://github.com/amazeeio/lagoon-cli/issues/189

**Comment by tobybellwood** [comment]
Unsuccessful operations should at least return unsuccessfully - what is returned is the subject of heated b/c conversations!

**Comment by shreddedbacon** [comment]
resolved in #3568
